pandora-server-directory/src/account/accountRelationship.ts

```diff
@@ -99,6 +99,48 @@ export class AccountRelationship {
 		return false;
 	}
 
+	/**
+	 * Checks if this account's profile is visible to the passed account
+	 * @param queryingAccount - The account attempting to access this account's profile
+	 * @returns - If the access should be allowed
+	 */
+	public async profileVisibleTo(queryingAccount: Account): Promise<boolean> {
```

Comment:
this is really similar to `canReceiveDM(from: Account)` function

could we merge it somehow, only difference is the moderator part and the allowDirectMessagesFrom, which I feel wouldn't be an issue if profile would be visible to all in that case

dms also have username, maybe we could use displayName there too

Reply:
I simplified the "characters share a room" bit by pulling it out into separate function. Otherwise I don't feel the similarity is a problem.
About just letting anyone see - I would prefer not to, but in the end it is just a preference.

For `displayName` - currently it is not configurable and just uses username;
I plan to go through existing code after this PR is merged and replace all cases where username is exposed directly with fully configurable display name. I'm not even sure yet if it should default to the username or to something else, e.g. `User <id>`.